src/master.rs

```diff
@@ -517,9 +517,97 @@ impl Master {
         Ok(time)
     }
 
+    pub fn foe_read(&mut self, idx: u16, name: String) -> Result<Vec<u8>> {
+        let file_name = string_to_foe_name(name);
+        // FIXME: this is the same as in the c-implementation. Should read in chunks instead of a
+        // fixed size buffer.
+        const FOE_SIZE: usize = 0x8800;
```

Comment:
Is this the maximum file size?

Reply:
Actually, I don't know. I just took the same size as in the c impl: https://gitlab.com/etherlab.org/ethercat/-/blob/master/tool/CommandFoeRead.cpp#L118. I dont' have access to the ethercat spec right now, but will have a look and try to find out where this number is coming from.

Reply:
So I couldn't really find anything for this and I don't think there is an actual maximum file size.

We could use a const generic `FOE_SIZE` like so

```rust
    pub fn foe_read<const FOE_SIZE: usize>(&mut self, idx: SlavePos, name: &str) -> Result<[i8; FOE_SIZE]> {
                let mut buf: Vec<u8> = vec![0; FOE_SIZE];
                ...
```

to allow at least some flexibility. But then the callee needs to know about the file size in advance.
We could also keep the `Vec<u8>` as return type and use `FOE_SIZE_LIMIT` or so as an upper bound for the file size?

Reply:
I'd like to keep the `Vec`; I assume that it's possible you don't know the exact size in advance. Is it possible to read in chunks?

Reply:
Yes, I'd also keep `Vec`.
I had a look at the implementation in the master: https://gitlab.com/etherlab.org/ethercat/-/blob/stable-1.5/master/ioctl.c#L3987, it allocates a buffer of 10000 bytes and reads the foe into that buffer. The buffered read/read in chunks would have to be implemented in the master. The `ioctl` call is already unbuffered and just provides you with the full package. If the provided buffer is to small it fails with `EOVERFLOW` (https://gitlab.com/etherlab.org/ethercat/-/blob/stable-1.5/master/ioctl.c#L4040)

I think it makes sense to also use 10000 bytes for now. What do you think?

Reply:
Agreed!